### ISSUE TITLE
parse-csv() - simplify output

### ISSUE THREAD

**Issue by michaelhkay** [qt4cg/qtspecs#1052]
Currently parse-csv produces a structure like this:

```
map {
  "columns": map {
      "names": map{"one":1, "two":2},
      "fields": ("one", "two")
   }
  "rows":  (
     map{
        "fields": ("aaa", "bbb"),
        "field" fn($col){$this?fields[$col]}
    },
   map{
        "fields": ("ccc", "ddd"),
        "field" fn($col){$this?fields[$col]}
    }
 )
}
```
There are a number of ways this could be improved.

- The structure is needlessly different from the return value of maps-to-arrays(). Users will get confused between the two representations, and will find it hard to switch from one to the other. For example, one delivers rows as sequences, the other delivers rows as arrays.
- There are too many levels in the structure; the expressions to select within it are unnecessarily complicated, and users will get poor diagnostics when they get it wrong.
- In this example (with two columns) for each row there is one map, one sequence, one function, and two strings - five values in all. The output of `csv-to-arrays` has only three objects (one array and two strings). However hard an optimized implementation tries to reduce the overhead, the space occupied by a million-row parsed CSV is likely to be larger than needed.
- The use of sequences rather than arrays means that no JSON-serialization of the structure is possible

I propose using a flatter structure, like this pseudo-code sketch:

```
map {
      "column-index": map{"one":1, "two":2},
      "columns": ["one", "two"]
      "rows": (
         ["aaa", "bbb"],
         ["ccc", "ddd"]
     )
     "get": fn($row, $col){$this?rows[$row]($col)}
     "size": fn(){count($this?rows)}
}
```
This doesn't meet all the objections outlined above; for example it represents rows as a sequence of arrays, which is consistent with `csv-to-arrays`, but not JSON-serializable. But I think it's a considerable improvement.

**Comment by ChristianGruen** [comment]
> ```xquery
> map {
>       "column-index": map{"one":1, "two":2},
>       "columns": ["one", "two"]
>       "rows": (
>          ["aaa", "bbb"],
>          ["ccc", "ddd"]
>      )
>      "get": fn($row, $col){$this?rows[$row]($col)}
>      "size": fn(){count($this?rows)}
> }
> ```

I’m definitely in line with the proposed structure. It will allow us to do traversals as simple as…

```xquery
for $row at $r in $csv?rows
for member $field at $c in $row
return `{$r}/{$c}: {$field}`
```

…and it’s very similar to our own CSV format, so I may be biased ;·)

> This doesn't meet all the objections outlined above; for example it represents rows as a sequence of arrays, which is consistent with `csv-to-arrays`, but not JSON-serializable. 

I’m convinced we should be able manage to serialize sequences with JSON (see https://github.com/qt4cg/qtspecs/issues/576#issuecomment-1768087310).

**Comment by michaelhkay** [comment]
Closed by #1066
